app/src/test/js/mockService.js

```diff
@@ -0,0 +1,26 @@
+var mockService = {
+    getSyncKeysTree : function() {
+        deferred = $q.defer();
```

Comment:
It looks like this is a global object (`deferred`) that is shared. If you omit `var` on a declaration, it defines it globally (weird javascript quirk). I think you mean to put `var ` in front of these, otherwise you are kind of sharing the defer object and it's a little weird.